src/dashboards/components/AutoRefreshOverlay.tsx

```diff
@@ -0,0 +1,141 @@
+
+// Context
+import AutoRefreshContextProvider, {
+  AutoRefreshContext,
+} from 'src/dashboards/components/RefreshContext'
+
+import './AutoRefresh.scss'
+
+const INACTIVITY_ARRAY = [...Array(25).keys()].map(num => num.toString())
+INACTIVITY_ARRAY[0] = 'None'
+
+export const AutoRefreshForm: FC = () => {
+  const {onClose} = useContext(OverlayContext)
+  const {state, dispatch: setRefreshContext, activateAutoRefresh} = useContext(
+    AutoRefreshContext
+  )
+
+  const handleChooseAutoRefresh = milliseconds => {
+    const status =
+      milliseconds === 0 ? AutoRefreshStatus.Paused : AutoRefreshStatus.Active
+    setRefreshContext({
+      type: 'SET_REFRESH_MILLISECONDS',
+      refreshMilliseconds: {status, interval: milliseconds},
+    })
+  }
+
+  // Because we are not using the manual refresh in the dropdown, this function is a no-op
+  const noop = useCallback((): void => {}, [])
+  return (
+    <Overlay.Container maxWidth={500} testID="auto-refresh-overlay">
+      <Overlay.Header title="Configure Auto Refresh" onDismiss={onClose} />
+      <Grid>
+        <Grid.Column className="refresh-form-column">
+          <div className="refresh-form-container">
+            <span>Select Refresh Frequency: </span>
+            <AutoRefreshDropdown
+              onChoose={handleChooseAutoRefresh}
+              onManualRefresh={noop}
+              selected={state.refreshMilliseconds}
+              showManualRefresh={false}
+            />
+          </div>
+          <div className="refresh-form-container">
+            <span className="refresh-form-container-child">Until: </span>
+            <TimeRangeDropdown
+              timeRange={state.duration}
+              onSetTimeRange={(timeRange: CustomTimeRange) =>
+                setRefreshContext({type: 'SET_DURATION', duration: timeRange})
+              }
+              singleDirection="upper"
+              className="refresh-form-container-child"
+            />
+          </div>
+          <div className="refresh-form-container">
+            <span className="refresh-form-container-child">
+              Inactivity Timeout:{' '}
+            </span>
+            <div
+              className={
+                state.inactivityTimeout === 'None'
+                  ? 'refresh-form-container-child inactive'
+                  : 'refresh-form-container-child active'
+              }
+            >
+              <SelectDropdown
+                options={INACTIVITY_ARRAY}
+                selectedOption={state.inactivityTimeout}
+                onSelect={(timeout: string) =>
+                  setRefreshContext({
+                    type: 'SET_INACTIVITY_TIMEOUT',
+                    inactivityTimeout: timeout,
+                  })
+                }
+                buttonColor={ComponentColor.Default}
+                testID="inactivity-timeout-dropdown"
+              />
+              {state.inactivityTimeout !== 'None' && (
```

Comment:
this isn't a design paradigm we have introduced yet. i'd err to figuring out how to use the `10d 1h 30m` time format that we use everywhere else in the app in the name of consistency, but get the sign off of daniel and russ that it's ok to introduce time input in a different way, and that this is the way

Reply:
Hey @drdelambre, thanks for flagging this! Appreciate you hitting the brakes here and design input.

 For this usecase, I think the use of the two drop down (one for number, and one for unit of time) makes sense. While it is a new paradigm to our application, this is a familiar paradigm in modern(and not modern) applications. 
 
 cc @mavarius